cpp/arcticdb/storage/config_cache.hpp

```diff
@@ -111,18 +110,16 @@ class ConfigCache {
         for(auto& [storage_name, config] : storages) {
             storage_configs_.try_emplace(StorageName(storage_name), config);
         }
-        auto default_storages = config_resolver_->get_default_storages(environment_name_);
-        for(auto& [storage_name, config] : default_storages) {
-            if (storage_configs_.find(storage_name) == storage_configs_.end()) {
-                config_resolver_->add_storage(environment_name_, storage_name, config);
-                storage_configs_.try_emplace(StorageName(storage_name), config);
-            }
+        auto native_storages = config_resolver_->get_native_storages(environment_name_);
+        for(auto& [storage_name, config] : native_storages) {
+            native_storage_configs_.try_emplace(StorageName(storage_name), config);
         }
     }
 
     EnvironmentName environment_name_;
     std::unordered_map<LibraryPath, LibraryDescriptor> descriptor_map_;
     std::unordered_map<StorageName, arcticdb::proto::storage::VariantStorage> storage_configs_;
```

Comment:
Why can't we replace `storage_configs_` with `native_storage_configs_`? Why do we need both? Can't we just change the type of `storage_configs_`?